### PR TITLE
(maint) Add supercronic binary to container for k8s

### PR DESCRIPTION
**Pull request by underscorgan**
Please add all notable changes to the "Unreleased" section of the CHANGELOG.

This supersedes #1043 and includes some general cleanup and brings the r10k dockerfiles/automation up to date with other projects

### CODE REVIEW

**Comment by underscorgan** [comment]
fyi @Xtigyro

**Comment by Xtigyro** [comment]
@underscorgan Awesome - thank you very much!

File: docker/r10k/Dockerfile [hunk 1]
```diff
@@ -32,24 +37,29 @@ COPY docker/r10k/docker-entrypoint.d /docker-entrypoint.d
...
+RUN chmod a+x /adduser.sh /docker-entrypoint.sh && \
 # Add a puppet user to run r10k as for consistency with puppetserver
     /adduser.sh && \
-    chmod +x /docker-entrypoint.sh && \
     chown -R puppet: /docker-entrypoint.d /docker-entrypoint.sh && \
     apk add --no-cache ruby openssh-client git ruby-rugged curl ruby-json ruby-etc && \
     gem install --no-doc /r10k.gem && \
-    rm -f /r10k.gem
+    rm -f /r10k.gem && \
+    curl --fail --silent --show-error --location --remote-name "$supercronic_url" && \
```
Comment (Iristyle) [inline docker/r10k/Dockerfile:58]: A bit of this I would do differently for buildkit caching (i.e. use `ADD` to pull the release file)... but I need to put up a PR to merge together the 2 Dockerfiles with stages, so I'll address it there. Should be able to get rid of the `pipefail` at the same time as well)

**Review by Iristyle — DISMISSED** [review]

**Review by nwolfe — DISMISSED** [review]

**Review by Iristyle — DISMISSED** [review]

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -10,13 +10,18 @@ jobs:
...
+      # Still need these env vars for README publishing
       DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
       DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
 
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@master
+      - uses: azure/docker-login@v1
+        with: # This doesn't seem to work unless we point directly to the secrets
```
Comment (Iristyle) [inline .github/workflows/docker.yml:22]: Yeah, I don't see anything special in the code - https://github.com/Azure/docker-login/blob/master/src/login.ts

**Comment by underscorgan** [comment]
@adrienthebo @dhollinger @puppetlabs/puppetserver-maintainers want to take a look?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -28,5 +33,24 @@ jobs:
       - name: Publish container
         working-directory: docker
         run: |
-          docker login -u "$DOCKERHUB_USERNAME" -p "$DOCKERHUB_PASSWORD"
           make publish
+      - name: Build release container
```
Comment (mwaggett) [inline .github/workflows/docker.yml:37]: Will these get triggered the same way as the dev/non-release builds?

Comment (underscorgan) [reply .github/workflows/docker.yml:37]: yep, this also runs on every commit, but just realized this container is lacking the logic to only build a new release container if one hasn't already been published for the latest version

Comment (mwaggett) [reply .github/workflows/docker.yml:37]: okay cool, yeah, I was wondering what was preventing it from building release containers for dev versions

**Review by mwaggett — APPROVED** [review]

**Review by Iristyle — APPROVED** [review]

**Pull request merged**